### PR TITLE
DCJ-453: Use GH secrets instead of vault

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Partially addresses https://broadworkbench.atlassian.net/browse/DCJ-453

~Requires: https://github.com/broadinstitute/terraform-ap-deployments/pull/1608~ (merged)

Manual GHA runs: 
* Staging Smoke tests: https://github.com/DataBiosphere/jade-data-repo/actions/runs/9603279556
* Unit, smoke, connected, integration tests: https://github.com/DataBiosphere/jade-data-repo/actions/runs/9611459203

### CODE REVIEW

**Review by okotsopoulos — APPROVED** [review]
Looks great -- from my vantage point, these two workflows will no longer talk to Vault directly at all after this PR :tada:

Looks like `cherry-pick-image.yaml` will be exercised on the merge of this PR to dev.  If you'd like to manually trigger Staging smoke tests off of this feature branch, that could be a good way to confirm that this change is working as expected:

https://github.com/DataBiosphere/jade-data-repo/actions/workflows/staging-smoke-tests.yaml

**Comment by rushtong** [comment]
> Looks great -- from my vantage point, these two workflows will no longer talk to Vault directly at all after this PR 🎉
> 
> Looks like `cherry-pick-image.yaml` will be exercised on the merge of this PR to dev. If you'd like to manually trigger Staging smoke tests off of this feature branch, that could be a good way to confirm that this change is working as expected:
> 
> https://github.com/DataBiosphere/jade-data-repo/actions/workflows/staging-smoke-tests.yaml

Yes, that is correct. At one point, we did the same thing in DUOS-land and were asked to move away from it. After this, all GHA vault usage will be specified in the same TF location so we can more easily standardize the migration to GSM.

**Review by fboulnois — APPROVED** [review]
👍

**Comment by rushtong** [comment]
~See https://github.com/DataBiosphere/jade-data-repo/pull/1718 for a replacement PR that hopefully doesn't get stuck in GHA limbo.~

File: .github/workflows/cherry-pick-image.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Note that this action is called from the UI build process:
+# https://github.com/DataBiosphere/jade-data-repo-ui/blob/develop/.github/workflows/dev-image-update.yaml#L95
+# Changes merged here require a version bump in that action.
```
Comment (okotsopoulos) [inline .github/workflows/cherry-pick-image.yaml:3]: Thanks very much for adding this!

File: .github/workflows/int-and-connected-test-run.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ on:
       - '**.gradle'
       - 'Dockerfile'
       - 'datarepo-clienttests/**'
-      - '.github/workflows/int-and-connected-test-run.yml'
+      - '.github/workflows/**'
```
Comment (okotsopoulos) [inline .github/workflows/int-and-connected-test-run.yml]: I am not convinced of the value or need to run int / connected tests for unrelated workflow changes, outside of making a required action for merge go green.  In those cases, I would be comfortable with merging the PR as an admin, but am curious what your thoughts are.

**Review by okotsopoulos — APPROVED** [review]
Looks good -- I'm not sure about the int / connected test suite trigger change, though.

File: .github/workflows/int-and-connected-test-run.yml (hunk 2 above)
Comment (rushtong) [reply .github/workflows/int-and-connected-test-run.yml]: I'm not convinced either - my goal was to see if I was understanding this correctly and I think I do. I am generally opposed to special PRs that are allowed to skip through required CI checks - it just adds to the degree of tribal knowledge necessary to maintain a repo. There should be a way to check that if an action in't run due to a `paths` exclude, then it should not be marked as required. That's essentially my next test.

**Comment by rushtong** [comment]
Proceeding with a bypass merge - required actions that aren't run are specifically excluded from running in this case due to file path exclusions.
